src/tests/multihost/alltests/test_krb_access_provider.py

```diff
@@ -0,0 +1,253 @@
+from constants import ds_instance_name
+
+
+@pytest.fixture(scope='class')
+def custom_setup(session_multihost, setup_sssd_krb, create_posix_usersgroups, krb_connection_timeout):
+    """ Added neccessary sssd domain parameters """
+    tools = sssdTools(session_multihost.client[0])
+    sssd_params = {'sbus_timeout': 30,
+                   'services': "nss, pam",
+                   'config_file_version': 2}
+    tools.sssd_conf('sssd', sssd_params)
+    domain_section = f'domain/{ds_instance_name}'
+    domain_params = {'access_provider': 'krb5',
+                     'use_fully_qualified_names': 'False',
+                     'override_homedir': '/home/%u'}
+    tools.sssd_conf(domain_section, domain_params)
+    tools.clear_sssd_cache()
+
+    cmds = ["mkdir -p /home/foo3",
+            "chown foo3: /home/foo3",
+            "chmod 700 /home/foo3",
+            "truncate -s 0 /home/foo3/.k5login",
+            "chown foo3: /home/foo3/.k5login",
+            "restorecon -v /home/foo3/.k5login"]
+    for command in cmds:
+        cmd = session_multihost.client[0].run_command(command)
+        assert cmd.returncode == 0
+
+
+@pytest.mark.krbaccessprovider
+@pytest.mark.usefixtures('custom_setup')
+class TestKrbAccessProvider(object):
+    """
+    This is test case class for krb_access_provider suite
+
+    Test access provision by kerberos on the basis of .k5login file.
+    Test for empty k5login file and test for presence of single or
+    multiple kerberos principal in the file, test for deleted user with
+    principals present in the file.
+    """
+    @staticmethod
+    def test_0001_k5login_empty(multihost):
+        """
+        :title: Create an empty .k5login file in the user's home directory
+        :id: e84bd024-669f-40d0-8833-edbc1be48e8e
+        :setup:
+          1. Clear the secure log, clear cache and restart sssd.
+        :steps:
+          1. Authenticate the ldapusers from the client
+          2. Check the secure log for expected messages.
+          3. Check the sssd domain log for expected messages.
+        :expectedresults:
+          1. Secure Log contains the expected lines:
+             pam_sss(sshd:auth): authentication success for foo3
+             pam_sss(sshd:account): Access denied for user foo3
+
+             pam_sss(sshd:auth): authentication success for foo4
+             Accepted password for foo4
+          2. SSSD Domain Log contains the expected lines:
+             Access denied for user [foo3
+             Access allowed for user [foo4
+        :teardown:
+          1. Clear the .k5login file.
+        """
+        multihost.client[0].run_command("truncate -s 0 /var/log/secure")
+        client = sssdTools(multihost.client[0])
+        client.clear_sssd_cache()
+
+        users = ['foo3', 'foo4']
+        ssh = client.auth_from_client(users[0], 'Secret123')
+        assert ssh == 10, "foo3 successfully logged In"
+        ssh = client.auth_from_client(users[1], 'Secret123')
```

Comment:
Please change the second ssh variable to ssh2 and move the asserts after multihost.client[0].run_command("truncate -s 0 /home/foo3/.k5login") so the file is cleaned up even when the asserts/test fails.

Consider it a general advice to do any cleanup (ip possible) before running asserts.